tests/unit/select_connection_ioloop_tests.py

```diff
@@ -35,10 +46,13 @@ def setUp(self):
         self.ioloop = select_connection.IOLoop()
 
     def tearDown(self):
-        self.ioloop.remove_timeout(self.fail_timer)
+        if hasattr(self, 'fail_timer'): # some test don't use this timer.
```

Comment:
Don't need any of this `remove_timeout` logic here at all. Instead, use addCleanup to register cleanup call after creating fail timer in `start()`. This make the code more readable. See my comment in `start()`